packages/controller-config/src/schemav2.ts

```diff
@@ -66,11 +66,18 @@ export const ControllerConfigSchemaV2 = yup
     infrastructureName: yup.string().required(),
     aws: yup.mixed<AWSConfig | undefined>(),
     gcp: yup.mixed<GCPConfig | undefined>(),
-    controllerTerminated: yup.bool().default(false)
+    controllerTerminated: yup.bool().default(false),
+    installerMetadata: yup.object({
+      lastInstallerVersion: yup.string(),
+    }),
```

Comment:
Because this adds a new field to the v2 schema it'll fail during controller configuration upgrades. The reason being that [upgradeControllerConfigMapToLatest](https://github.com/opstrace/opstrace/blob/e4c1c289ecfb917aca62143d6c942bd7ceffe265/packages/controller-config/src/schema.ts#L79-L107) will not be able to validate the (now old) v2 schema configuration on a running instance because the `installerMetadata` is now a required field.

One option is to make `installerMetadata` not required the other would be to introduce a new schema version (v3) and a `V2toV3` upgrade function that would set the new field.

Makes sense?

Reply:
I was concerned that adding a new field would break some contract. Typically when I think of changes that require a major version upgrade, I think of breaking changes to an existing contract; in this case, `installerMetadata` is a net-new field that isn't _required_ for any CLI functionality to work; however I could see some project component eventually relying on it's presence for all v2 configs when that isn't guaranteed if it remains optional.

How about schema v3alpha1, in case `installerMetadata`'s schema is changed soon based on new ideas? Is there a stabilization timeline you typically follow?

Reply:
> How about schema v3alpha1, in case installerMetadata's schema is changed soon based on new ideas? Is there a stabilization timeline you typically follow?

At this point, we don't have a schedule to handle these schema upgrades. AFAIK, there are no plans to touch the controller config map schema in the short term. You can go ahead and introduce `V3` then.

Reply:
Done.